lib/cordova-plugins/factory.js

```diff
@@ -0,0 +1,182 @@
+const logger = require('../Logger');
+
+/**
+ * plugin mock factory
+ * to create a proper abstraction for the test, the test does not neet to know whrere it's mocks come from !
+ *
+ * * How to mock a plugin
+ * ----------------------
+ * 1. Write its name, need to match the cordova plugin name and the actual mock filename, into the `wdi5: plugins: <pluginname>` config.
```

Comment:
put the exact cordova plugin name and its' mock countpart into: wdi5: { plugins: { <pluginName>: {} } }